src/target/stm32mp15.c

```diff
@@ -0,0 +1,154 @@
+/*
+ * This file is part of the Black Magic Debug project.
+ *
+ * Copyright (C) 2023 1BitSquared <info@1bitsquared.com>
+ * Written by ALTracer <tolstov_den@mail.ru>
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+/*
+ * This file implements STM32MP15 target specific functions for detecting
+ * the device and providing the XML memory map.
+ *
+ * References:
+ * RM0436 - STM32MP157 advanced Arm®-based 32-bit MPUs, Rev. 5
+ *   https://www.st.com/resource/en/reference_manual/rm0436-stm32mp157-advanced-armbased-32bit-mpus-stmicroelectronics.pdf
+ */
+
+#include "general.h"
+#include "target.h"
+#include "target_internal.h"
+#include "cortexm.h"
+
+/* Access from processor address space.
+ * Access via the debug APB is at 0xe0081000 over AP1. */
+#define DBGMCU_IDCODE      0x50081000U
+#define STM32MP15_UID_BASE 0x5c005234U
+
+#define DBGMCU_IDC (DBGMCU_IDCODE + 0U)
+#define DBGMCU_CR  (DBGMCU_IDCODE + 4U)
+#define DBGSLEEP   (1U << 0U)
+#define DBGSTOP    (1U << 1U)
+#define DBGSTBY    (1U << 2U)
+
+#define ID_STM32MP15x        0x5000U /* RM0436 */
+#define ID_STM32MP15x_ERRATA 0x450U
+
+typedef struct stm32mp15_priv {
+	uint32_t dbg_cr;
+} stm32mp15_priv_s;
+
+static bool stm32mp15_uid(target_s *target, int argc, const char **argv);
+static bool stm32mp15_cmd_rev(target_s *target, int argc, const char **argv);
+
+const command_s stm32mp15_cmd_list[] = {
+	{"uid", stm32mp15_uid, "Print unique device ID"},
+	{"revision", stm32mp15_cmd_rev, "Returns the Device ID and Revision"},
+	{NULL, NULL, NULL},
+};
+
+static void stm32mp15_detach(target_s *target)
+{
+	stm32mp15_priv_s *ps = (stm32mp15_priv_s *)target->target_storage;
```

Comment:
`ps` -> `priv` please - trying to avoid more single/two letter variable names.

Reply:
We were suggesting `priv` specifically (we're actually working on dropping the `_storage` elsewhere) as while the target member has storage in its name (accurate for that structure) the result of the cast is the private structure - not storage for that structure. Hence as it's called `stm32mp15_priv_s` -> `priv` for the variable name.

Reply:
Accepted pending rename in detach and in probe to `priv`.
Should I also factor out this dbg_cr manipulation at end of probe into a separate _attach to decouple sleep-inhibition from scans? (this will also use the priv)

Reply:
It'd definitely be more hygienic, and especially as this currently makes detaching one-shot with scan required to bring the part back into a good state to attach again. 👍🏼

Reply:
It seems like your pending fix for this never got pushed?

Reply:
priv_storage rename was pushed in 88cccfd4, and I kinda forgot about _attach. Will revisit soon. Should be trivial.